### DISCUSSION TITLE
Ingest wrfout NetCDFs

### DISCUSSION THREAD

**Discussion by Tuculca** [dtcenter/METplus#2139] [File I/O] [answered]
I would like to know what is the easiest way to use wrfout files in MET(plus), for example in PCPcombine.
I would REALLY like to do it using supported tools.

From an answer by @balbrightwpc I read:
"MET can read a few different flavors of NetCDF files:
- its internal format which is output from tools like gen_vx_mask or pcp_combine
- files following the [Climate Forecast (CF) convention](https://cfconventions.org/latest.html)
- WRF output post-processed by the WRF-interp utility"

I cannot find information about this WRF-interp utility, where can I find it?

Any suggestion would be very appreciated.
Thank you!

**Comment by Tuculca** [comment]
Sorry, maybe this is more appropriate in the Discussions section. In that case please delete the issue.

**Comment by jprestop** [comment, marked as answer]
Hi @Tuculca.  My apologies for such a delayed response.  Somehow this Discussion got lost and we're just now seeing it. 

I see that you'd like to better understand how you can best use wrfout files in METplus.  Thank you for looking at previous Discussions for an answer.  As you noted:

> MET can read a few different flavors of NetCDF files:
> - its internal format which is output from tools like gen_vx_mask or pcp_combine
> - files following the [Climate Forecast (CF) convention](https://cfconventions.org/latest.html)
> - WRF output post-processed by the WRF-interp utility

For your gridded model output, you have two options:
(1) Run the wrf-interp post-processing utility on your WRF output (linked on this MMM website: https://www2.mmm.ucar.edu/wrf/users/utilities/util.htm)
(2) Post-process your output using [UPP](http://www.dtcenter.org/community-code/unified-post-processor-upp) (the Unified Post Processor) which writes GRIB1 or GRIB2 output files. 

WRF-Interp does interpolate from native model levels to pressure levels, which is good, but it does not de-stagger the winds. And MET won't process the winds on the staggered dimensions.

So we recommend using UPP which destaggers the winds, interpolates to pressure levels, and optionally regrids the data.

That being said, if you don't care about the winds and are only interested in surface fields, MET can actually read the WRF output files directly. MET can read single levels of wrfout files directly if file_type = NETCDF_PINT is used.  For more information on file_type, you can search for file_type in this section of our documentation: https://met.readthedocs.io/en/latest/Users_Guide/config_options.html#fcst

Please let us know what other questions you have.

**Reply by jprestop** [reply]
Hi @Tuculca. I just wanted to follow up. Do you have any additional questions?

**Reply by Tuculca** [reply]
Sorry @jprestop I missed the answer, got no notification don't know why.
No additional questions, your answer totally addresses my question, thank you!

**Reply by jprestop** [reply]
That's ok @Tuculca.  Thank you for letting us know.  I'll go ahead and mark this as answered and lock this Discussion.  Please feel free to post a new Discussion with any additional questions.
